File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.Arrays;

class Solution {
    
    public int solution(int[] people, int limit) {
        
        Arrays.sort(people);
        
        int answer = 0;
        int num = people.length - 1;
        for(int i = 0; i <= num; i++, answer++)
            while(num > i && people[i] + people[num--] > limit)
                answer++;
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    people.sort()
    answer = 0
    left = 0
    right = len(people) - 1
    
    while left <= right:
        if people[left] + people[right] <= limit:
            left += 1
        right -= 1
        answer += 1
    
    return answer